### PR TITLE
output consistency: cleanup: remove unnecessary links

### PR DESCRIPTION
**Pull request by nrainer-materialize**

### CODE REVIEW

File: misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py [hunk 1]
```diff
@@ -231,7 +225,6 @@ def is_function_taking_shortcut(expression: Expression) -> bool:
 
         for expression in expressions:
             if expression.contains(is_function_taking_shortcut, True):
-                # see https://github.com/MaterializeInc/materialize/issues/17189
                 return True
```
Comment (def-) [inline misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py:228]: Could we keep a comment here like `TODO: Remove when #17189 is fixed`?

File: misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py [hunk 2]
```diff
@@ -248,7 +241,6 @@ def is_null_expression(expression: Expression) -> bool:
             if expression.contains(is_null_expression, True):
                 # Constant folding takes shortcuts when it can infer that an expression will be NULL or not
                 # (e.g., `chr(huge_value) = NULL` won't be fully evaluated)
-                # see https://github.com/MaterializeInc/materialize/issues/17189
```
Comment (def-) [inline misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py:251]: same

**Review by def- — CHANGES_REQUESTED** [review]

File: misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py:228]: It is still there as a `YesIgnore` where the return value is consumed. Therefore, I don't think it is necessary.

File: misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py (hunk 2 above)
Comment (nrainer-materialize) [reply misc/python/materialize/output_consistency/ignore_filter/inconsistency_ignore_filter.py:251]: also same as above 🙂

**Review by def- — APPROVED** [review]

**Pull request merged**